src/components/cardcollection/CardCollectionMap.java

```diff
@@ -0,0 +1,79 @@
+package components.cardcollection;
+
+import components.map.Map;
+import components.map.Map1L;
+
+/**
+ * {@code CardCollection} represented as {@link components.map} and
+ * {@code String} with implementations of primary methods.
+ */
+public class CardCollectionMap extends CardCollectionSecondary {
+    /**
+     * Private members -- -----------------------------------------------------.
+     */
+
+    //The card name of this card.
+    private String card;
+
+    /**
+     * The map that stores the card values of this card. Map Pair: Key: The date
+     * as a string of the value of this card. Value: The value as an integer of
+     * this card at the date (Key).
+     */
+    private Map1L<String, Integer> cardValues;
```

Comment:
There is probably a data structure in Java you can use for date (e.g., DateTime).